### PR TITLE
Sakib - React Kanban Board - Complete project with dark theme support

### PR DESCRIPTION
**Pull request by shadman0711**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Users can add/move items to the board

## What was done
* Drag and drop added
* Dark theme added
* Context API added

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: react-kanban/src/context/backlog/BacklogContext.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { createContext, useReducer } from 'react';
```
[the remark this answers is not in the document]
Comment (shadman0711) [reply react-kanban/src/context/backlog/BacklogContext.js]: Changed file name in commit [8779403](https://github.com/CraftsmenLtd/coe-batch-q322/pull/276/commits/87794035dab8d01e6d27098f7f2e9346aef8adf0)

File: react-kanban/src/context/backlog/backlogReducer.js [hunk 2]
```diff
@@ -30,11 +30,13 @@ export const backlogReducer = (state, action) => {
     const { type, payload } = action;
 
     switch (type) {
-        case "ADD_TO_BACKLOG":
+        case "UPDATE_BOARD":
```
[the remark this answers is not in the document]
Comment (shadman0711) [reply react-kanban/src/context/backlog/backlogReducer.js]: Changed in commit [8779403](https://github.com/CraftsmenLtd/coe-batch-q322/pull/276/commits/87794035dab8d01e6d27098f7f2e9346aef8adf0)

File: react-kanban/src/components/container/boards/Boards.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
-            <Board boardName='Ready' boardType='ready' />
-            <Board boardName='Done' boardType='done' />
-        </BoardsStyled>
+        <DndProvider backend={HTML5Backend}>
+            <BoardsStyled>
+                <Board boardName='Backlog' boardType='backlog' showInputForm />
+                <Board boardName='In Progress' boardType='inProgress' />
+                <Board boardName='In Review' boardType='inReview' />
+                <Board boardName='Ready' boardType='ready' />
+                <Board boardName='Done' boardType='done' />
```
[the remark this answers is not in the document]
Comment (shadman0711) [reply react-kanban/src/components/container/boards/Boards.js]: Added enums in commit [f5e6cbf](https://github.com/CraftsmenLtd/coe-batch-q322/pull/276/commits/f5e6cbffbe66c7e456136226a9ac589a0214324a)

File: react-kanban/src/components/container/boards/board/boardItem/BoardItem.js [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
 import { BoardItem as BoardItemStyled } from './BoardItem.styled';
+import { useDrag } from "react-dnd";
+import { ThemeContext } from '../../../../../context/theme/ThemeContext';
 
-export function BoardItem({ id, text }) {
+export function BoardItem({ item, board }) {
+    const { isDarkMode } = useContext(ThemeContext);
+
+    const [{ isDragging }, drag] = useDrag(() => ({
+        type: "card",
```
[the remark this answers is not in the document]
Comment (shadman0711) [reply react-kanban/src/components/container/boards/board/boardItem/BoardItem.js]: Added enums in commit [f5e6cbf](https://github.com/CraftsmenLtd/coe-batch-q322/pull/276/commits/f5e6cbffbe66c7e456136226a9ac589a0214324a)

File: react-kanban/src/context/backlog/KanbanContext.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import { createContext, useReducer } from 'react';
```
Comment (tusher-A) [inline react-kanban/src/context/backlog/KanbanContext.js:1]: Although you filename is `KanbanContext`, you are keeping it in a folder called `backlog`. Does that make sense to you?

**Review by tusher-A — CHANGES_REQUESTED** [review]
A minor change is requested. Please follow through.

File: react-kanban/src/context/backlog/KanbanContext.js (hunk 5 above)
Comment (shadman0711) [reply react-kanban/src/context/backlog/KanbanContext.js]: I've changed the directory name in commit [2dbc526](https://github.com/CraftsmenLtd/coe-batch-q322/pull/276/commits/2dbc5260085d045e7384a4f6a7537ec743c86112)

**Review by tusher-A — APPROVED** [review]
LGTM.

**Pull request merged**